core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -97,7 +87,9 @@ class KryoSerializer(conf: SparkConf)
       // Use the default classloader when calling the user registrator.
       Thread.currentThread.setContextClassLoader(classLoader)
       // Register classes given through spark.kryo.classesToRegister.
-      classesToRegister.foreach { clazz => kryo.register(clazz) }
+      classesToRegister.split(',')
```

Comment:
Can we do the splitting and filtering during initialization so that we don't incur them for each task?